l1-contracts/test/foundry/unit/concrete/Bridgehub/experimental_bridge.t.sol

```diff
@@ -0,0 +1,744 @@
+        // to the same function in the mailbox, we will mock the call to the mailbox to return true and see if it works.
+        vm.mockCall(
+            address(mockChainContract),
+            abi.encodeWithSelector(
+                mockChainContract.proveL2MessageInclusion.selector,
+                mockBatchNumber,
+                mockIndex,
+                l2Message,
+                mockProof 
+            ),
+            abi.encode(true)
+        );
+
+        assertTrue(bridgeHub.proveL2MessageInclusion(mockChainId,mockBatchNumber,mockIndex,l2Message,mockProof));
+        vm.clearMockedCalls();
+    }
+
+    function test_proveL2LogInclusion(
+        uint256 mockChainId,
+        uint256 mockBatchNumber,
+        uint256 mockIndex,
+        bytes32[] memory mockProof,
+        uint8 randomL2ShardId,
+        bool randomIsService,
+        uint16 randomTxNumInBatch,
+        address randomSender,
+        bytes32 randomKey,
+        bytes32 randomValue
+    ) public {
+        mockChainId = _setUpStateTransitionForChainId(mockChainId);
+
+        // Now the following statements should be true as well:
+        assertTrue(bridgeHub.stateTransitionManager(mockChainId) == address(mockSTM));
+        assertTrue(bridgeHub.getStateTransition(mockChainId) == address(mockChainContract));
+
+        // Creating a random L2Log::l2Log so that we pass the correct parameters to `proveL2LogInclusion`
+        L2Log memory l2Log = _createMockL2Log(
+            randomL2ShardId,
+            randomIsService,
+            randomTxNumInBatch,
+            randomSender,
+            randomKey,
+            randomValue
+        );
+
+        // Since we have used random data for the `bridgeHub.proveL2LogInclusion` function which basically forwards the call
+        // to the same function in the mailbox, we will mock the call to the mailbox to return true and see if it works.
+        vm.mockCall(
+            address(mockChainContract),
+            abi.encodeWithSelector(
+                mockChainContract.proveL2LogInclusion.selector,
+                mockBatchNumber,
+                mockIndex,
+                l2Log,
+                mockProof 
+            ),
+            abi.encode(true)
+        );
+
+        assertTrue(bridgeHub.proveL2LogInclusion(mockChainId,mockBatchNumber,mockIndex,l2Log,mockProof));
+        vm.clearMockedCalls();
+    }
+
+    function test_proveL1ToL2TransactionStatus(
+        uint256 randomChainId,
+        bytes32 randomL2TxHash,
+        uint256 randomL2BatchNumber,
+        uint256 randomL2MessageIndex,
+        uint16 randomL2TxNumberInBatch,
+        bytes32[] memory randomMerkleProof,
+        bool randomResultantBool
+    ) public {
+        randomChainId = _setUpStateTransitionForChainId(randomChainId);
+
+        TxStatus txStatus;
+
+        if (randomChainId % 2 == 0) {
+            txStatus = TxStatus.Failure;
+        } else {
+            txStatus = TxStatus.Success;
+        }
```

Comment:
maybe you should fuzz `txStatus` value too? If fuzzer can't set `TxStatus` value you can use boolean and then transform it to `TxStatus` in the code.

Reply:
Done.